From f7d4c02fd522a107812aa65b8c1b8279af2ee263 Mon Sep 17 00:00:00 2001
From: Andy Grover <agrover@redhat.com>
Date: Tue, 7 Oct 2014 11:38:16 -0700
Subject: [PATCH] iscsiuio: Resolve strict aliasing issue in
 iscsiuio/src/unix/nic.c

Break up the expression and use a local variable.

Signed-off-by: Andy Grover <agrover@redhat.com>
---
 iscsiuio/src/unix/nic.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/iscsiuio/src/unix/nic.c b/iscsiuio/src/unix/nic.c
index e00e8f2..d56a9da 100644
--- a/iscsiuio/src/unix/nic.c
+++ b/iscsiuio/src/unix/nic.c
@@ -999,11 +999,13 @@ int process_packets(nic_t *nic,
 		vlan_id = pkt->vlan_tag & 0xFFF;
 		if ((vlan_id == 0) ||
 		    (NIC_VLAN_STRIP_ENABLED & nic->flags)) {
-			type = ntohs(ETH_BUF(pkt->buf)->type);
+			struct uip_eth_hdr *hdr = ETH_BUF(pkt->buf);
+			type = ntohs(hdr->type);
 			pkt->network_layer = pkt->data_link_layer +
 					     sizeof(struct uip_eth_hdr);
 		} else {
-			type = ntohs(VLAN_ETH_BUF(pkt->buf)->type);
+			struct uip_vlan_eth_hdr *hdr = VLAN_ETH_BUF(pkt->buf);
+			type = ntohs(hdr->type);
 			pkt->network_layer = pkt->data_link_layer +
 					     sizeof(struct uip_vlan_eth_hdr);
 		}
-- 
2.5.0

